src/python/WMCore/WMRuntime/Timestamps.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python
+"""
+_Timestamps_
+
+"""
+
+import getopt
+import logging
+import os
+import pickle
+import sys
+
+
+# script options
+options = {"reportFile=": "", "wmJobTime=": ""}
+
+
+def addTimestampMetrics(config, wmJobTime):
+    """
+    Adjust timestamp metrics of provied FJR data.
+
+    :param config: input FJR data
+    :param wmJobTime: total wall clock time of WM job in seconds
+    """
+    timing = config.section_('WMTiming')
+    timing.WMTotalWallClockTime = wmJobTime
+
+
+def main():
+    """
+    Main function of the script performs business logic:
+    - parse input arguments
+    - read provided input FJR
+    - adjust timestamp metrics of FJR data
+    - write out FJR
+    """
+    try:
+        opts, _ = getopt.getopt(sys.argv[1:], "", list(options.keys()))
+    except getopt.GetoptError as ex:
+        msg = "Error processing commandline args:\n"
+        msg += str(ex)
+        logging.error(msg)
+        sys.exit(1)
+
+    for opt, arg in opts:
+        if opt == "--reportFile":
+            reportFile = arg
+        if opt == "--wmJobTime":
+            wmJobTime = arg
+
+    # read content of given FJR report file
+    data = {}
+    with open(reportFile, 'rb') as istream:
+        data = pickle.load(istream)
+
+    # adjust FJR data with provided metrics
+    addTimestampMetrics(data, wmJobTime)
+
+    # write content of FJR back to report file
+    reportOutFile = reportFile + ".new"
+    msg = f"Adding wmJobTime metric to {reportFile}"
+    with open(reportOutFile, 'wb') as ostream:
+        logging.info(msg)
+        pickle.dump(data, ostream)
+
+    # if we successfully wrote reportOutFile we can swap it with input one
```

Comment:
I think we can remove this block and simply write the up-to-date content to the same pickle file name. Did you try this?

Reply:
No, I will preserve current functionality since it does few things:
- it checks that both new and current output file exists and have non-zero size
- and if it is not the case it will not overwrite existing pickle file which I think would be useful in debugging process, also the information in such pickle file will not be lost. Otherwise, we may over write file and lost its information which I doubt is desired behavior.

Reply:
And why do you think the file would be empty? If there is any error writing to the output file, then we would anyways hit an exception either at the pickle.dump or the os.rename. Renaming the file is one extra command that can hit a failure.